### PR TITLE
Refactor developer documentation

### PR DESCRIPTION
**Pull request by lbernick**
# Changes
This commit splits developer documentation into separate files, with a table of contents to help developers find docs more easily. It also reorganizes the results documentation to more clearly separate results documentation from results implementation (except in the case of a known issue with our implementation).

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- n/a Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- n/a Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- n/a Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/developers/api-versioning.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# API Versioning
+
+## Adding feature gated API fields
+
+We've introduced a feature-flag called `enable-api-fields` to the
+[config-feature-flags.yaml file](../../config/config-feature-flags.yaml)
+deployed as part of our releases.
+
+This field can be configured either to be `alpha` or `stable`. This field is
```
Comment (vdemeester) [inline docs/developers/api-versioning.md:9]: This isn't true anymore right ? There is `beta` now.

File: docs/developers/api-versioning.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+validated by Pipelines' webhook. Here too you'll need to ensure your new
+features aren't accidentally accepted when the feature gate suggests they
+shouldn't be. We've got a helper function,
+[`ValidateEnabledAPIFields`](../../pkg/apis/pipeline/v1beta1/version_validation.go),
+to make validating the current feature gate easier. Use it like this:
+
+```go
+requiredVersion := config.AlphaAPIFields
+// errs is an instance of *apis.FieldError, a common type in our validation code
+errs = errs.Also(ValidateEnabledAPIFields(ctx, "your feature name", requiredVersion))
```
Comment (vdemeester) [inline docs/developers/api-versioning.md:51]: I know this is not related to this PR, but do we have one for the beta value as well ?

Comment (lbernick) [reply docs/developers/api-versioning.md:51]: this helper can be used for beta API fields as well.

Comment (lbernick) [reply docs/developers/api-versioning.md:51]: it's here now https://github.com/tektoncd/pipeline/blob/main/pkg/apis/version/version_validation.go

File: docs/developers/taskruns.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+some is distributed across a number of fields of the `TaskRun's` `status`. For
+example, a `TaskRun's` `status.taskResults` is populated from the termination
+message.
+
+## Reserved directories
+
+### /workspace
+
+- `/workspace` - This directory is where [resources](#resources) and
+  [workspaces](#workspaces) are mounted.
```
[the remark this answers is not in the document]
Comment (lbernick) [reply docs/developers/taskruns.md:80]: reworded to make it clear that it's the volumes being mounted 👍

**Comment by lbernick** [comment]
thanks for the review @vdemeester, I've updated the PR!

**Review by vdemeester — APPROVED** [review]

**Review by afrittoli — COMMENTED** [review]
Thanks @lbernick 
Holding off lgtm for now in case @geriom has some comments.
Added this to the milestone so we merge it before v0.41.0 goes out.

/approve

**Comment by lbernick** [comment]
> Holding off lgtm for now in case @geriom has some comments.

Looks like this was merged but if Geri has any comments I am still happy to address them!

**Pull request merged**